src/docker/shared.rs

```diff
@@ -714,13 +714,24 @@ pub(crate) fn docker_mount(
     paths: &DockerPaths,
     mount_cb: impl Fn(&mut Command, &Path, &Path) -> Result<()>,
     mut store_cb: impl FnMut((String, String)),
+    msg_info: &mut MessageInfo,
 ) -> Result<()> {
+    let mut warned = false;
     for ref var in options
         .config
         .env_volumes(&options.target)?
         .unwrap_or_default()
     {
         let (var, value) = validate_env_var(var)?;
+        if value.is_none()
+            && !warned
+            && !var
+                .chars()
+                .all(|c| matches!(c, 'a'..='z' | 'A'..='Z' | '_' ))
+        {
+            msg_info.warn(format_args!("got volume of \"{var}\" which is not an environment variable name. the proper syntax is `volumes = [\"ENVVAR=/path/to/directory\"]`"))?;
```

Comment:
should this warn or error?

im fine with either

Reply:
I think it should warn. It's technically valid input to the Docker container, it just won't behave as expected.